src/main/java/org/spongepowered/api/service/command/SimpleCommandService.java

```diff
@@ -61,7 +62,7 @@
 
     private final PluginManager pluginManager;
     private final SimpleDispatcher dispatcher = new SimpleDispatcher();
-    private final Multimap<PluginContainer, CommandMapping> owners = HashMultimap.create();
+    private final Multimap<PluginContainer, ImmutableCommandMapping> owners = HashMultimap.create();
```

Comment:
Maybe, i can remove it if it is not liked (I used a second commit for that so i can revert it if it is not liked.) , but from my point of view the SimpleCommandService is a class that will not be subclasses and thus can be manipulated to match its real contents.

Its more or less an example how the changes to the API could be used. 

I also work on the server implementation, but there are currently some unrelated issues that prevent my IDE from showing any useful compile errors.